cmd/oracled/cmd/register_oracle.go

```diff
@@ -41,146 +37,148 @@ func registerOracleCmd() *cobra.Command {
+	}
 
-			event := oracleevent.NewApproveOracleRegistrationEvent()
+	log.Infof("register-oracle transaction succeed. height(%v), hash(%s)", resp.TxResponse.Height, resp.TxResponse.TxHash)
 
-			ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
-			defer cancel()
+	return nil
+}
 
-			txs, err := client.Subscribe(ctx, "", event.GetEventQuery())
-			if err != nil {
-				return err
-			}
+func generateMsgRegisterOracle(cmd *cobra.Command, conf *config.Config, oracleAccount *panacea.OracleAccount, trustedBlockInfo *panacea.TrustedBlockInfo) (*oracletypes.MsgRegisterOracle, error) {
+	// if node key exists, return error.
+	nodePrivKeyPath := conf.AbsNodePrivKeyPath()
+	if tos.FileExists(nodePrivKeyPath) {
+		buf := bufio.NewReader(os.Stdin)
+		ok, err := input.GetConfirmation("There is an existing node key. \nAre you sure to delete and re-generate node key?", buf, os.Stderr)
+		if err != nil || !ok {
+			log.Infof("Node key generation is canceled.")
+			return nil, err
+		}
+	}
 
-			errChan := make(chan error, 1)
+	// generate node key and its remote report
+	nodePubKey, nodePubKeyRemoteReport, err := generateAndSealedNodeKey(nodePrivKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate node key pair: %w", err)
+	}
 
-			for tx := range txs {
-				errChan <- event.EventHandler(tx)
-			}
+	report, err := enclave.VerifyRemoteReport(nodePubKeyRemoteReport)
+	if err != nil {
+		return nil, fmt.Errorf("failed to verification remoteReport. %w", err)
+	}
+	uniqueID := hex.EncodeToString(report.UniqueID)
 
-			err = <-errChan
-			if err != nil {
-				log.Infof("Error occurs while getting shared oracle private key. Please retrieve it via get-oracle-key cmd: %v", err)
-				return err
-			} else {
-				log.Infof("oracle private key is successfully shared. You can start oracle now!")
-				return nil
-			}
-		},
+	oracleEndpoint, err := cmd.Flags().GetString(flags.FlagOracleEndpoint)
+	if err != nil {
+		return nil, err
 	}
 
-	cmd.Flags().Int64(flags.FlagTrustedBlockHeight, 0, "Trusted block height")
-	cmd.Flags().String(flags.FlagTrustedBlockHash, "", "Trusted block hash")
-	cmd.Flags().String(flagOracleEndpoint, "", "endpoint of oracle")
-	cmd.Flags().String(flagOracleDescription, "", "description of oracle")
-	cmd.Flags().String(flagOracleCommissionRate, "0.1", "oracle commission rate")
-	_ = cmd.MarkFlagRequired(flags.FlagTrustedBlockHeight)
-	_ = cmd.MarkFlagRequired(flags.FlagTrustedBlockHash)
+	// request register oracle Tx to Panacea
+	oracleCommissionRateStr, err := cmd.Flags().GetString(flags.FlagOracleCommissionRate)
+	if err != nil {
+		return nil, err
+	}
 
-	return cmd
+	oracleCommissionRate, err := sdk.NewDecFromStr(oracleCommissionRateStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse oracleCommissionRate. input(%s). %w", oracleCommissionRateStr, err)
+	}
+
+	msgRegisterOracle := oracletypes.NewMsgRegisterOracle(
```

Comment:
Requires an argument like `commissionMaxRate` and `commissionMaxChangeRate`. I think you should pull the latest panacea.

Reply:
It has already been done in #26. Please refer to it.